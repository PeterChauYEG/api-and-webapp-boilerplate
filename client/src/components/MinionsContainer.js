import React, { Component } from 'react'

// import components
import Minion from './Minion'

class MinionsContainer extends Component {
  renderMinions(minions) {
  
    // handle no minions
    if (minions.length === 0 ) {
      return <p>There's no minions yet. You should add one!</p>
    }
    
    // map minion data to component
    return minions.map((minion, i) => {
      return <Minion minion={minion} key={i} />
    })
  }

  render() {
    const {
      minions,
    } = this.props;

    return (
      <div>
        {
          // render minions
          this.renderMinions(minions)
        }
      </div>
    )
  }
}

MinionsContainer.propTypes = {
  minions: React.PropTypes.array.isRequired,
}

export default MinionsContainer;
