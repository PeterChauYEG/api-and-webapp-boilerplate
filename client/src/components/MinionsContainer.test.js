import React from 'react'

// import test dependencies 
import renderer from 'react-test-renderer'

// import component
import MinionsContainer from './MinionsContainer'

// render test
test('renders without crashing', () => {
  
  // declare prop
  const minions = []
  
  // render component
  const component = renderer.create(
    <MinionsContainer 
      minions={minions}
      />
  )

  // render DOM tree
  const tree = component.toJSON()
  
  // assertion
  expect(tree).toMatchSnapshot()
  
});

// render minions
test('renders minions', () => {
  
  // declare prop
  const minions = [
    {
      null
    },
    {
      null
    },
    {
      null
    },    
  ]
  
  // render component
  const component = renderer.create(
    <MinionsContainer 
      minions={minions}
      />
  )

  // render DOM tree
  const tree = component.toJSON()
  
  // assertion
  expect(tree).toMatchSnapshot()
  
});