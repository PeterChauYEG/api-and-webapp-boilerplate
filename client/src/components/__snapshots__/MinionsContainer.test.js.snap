exports[`test renders minions 1`] = `
<div>
  <div
    className="Minion">
    <h2 />
    <h3 />
    <p />
  </div>
  <div
    className="Minion">
    <h2 />
    <h3 />
    <p />
  </div>
  <div
    className="Minion">
    <h2 />
    <h3 />
    <p />
  </div>
</div>
`;

exports[`test renders without crashing 1`] = `
<div>
  <p>
    There\'s no minions yet. You should add one!
  </p>
</div>
`;
